File: app/models/NextQuestionHandler.java
package models;

import models.questions.Question;
import play.libs.F.Option;

import com.fasterxml.jackson.databind.JsonNode;

public class NextQuestionHandler implements Handler {

	private QuestionAsker asker;
	private QuizMaster quizMaster;
	private BuzzerManager buzzerManager; 
	
	public NextQuestionHandler(QuizMaster quizMaster, QuestionAsker asker, BuzzerManager buzzerManager) {
		this.asker = asker;
		this.quizMaster = quizMaster;
		this.buzzerManager = buzzerManager;
	}
	
	@Override
	public void handle(Team team, JsonNode message) {
		Question question = asker.nextQuestion();
		buzzerManager.reset();
		quizMaster.notifyTeams(Option.Some(question));
	}
}

File: appKotlin/models/NextQuestionHandler.kt
package models

import com.fasterxml.jackson.databind.JsonNode

import java.util.Optional

class NextQuestionHandler(private val quizMaster: QuizMaster, private val asker: QuestionAsker, private val buzzerManager: BuzzerManager) : Handler {

    override fun handle(team: Team, message: JsonNode) {
        val question = asker.nextQuestion()
        buzzerManager.reset()
        quizMaster.teamRoster.values
                .stream()
                .forEach { t -> t.resetBuzzer() }

        quizMaster.notifyTeams(Optional.of<Any>(question))
        quizMaster.notifyAdmin()
    }
}